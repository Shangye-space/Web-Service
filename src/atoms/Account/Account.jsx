import React from 'react';

import styles from './Account.module.scss';

const Account = ({ isVisible, setVisibility, userData }) => {
    function formatName(name) {
        return name.substring(0, 1).toUpperCase() + name.substring(1);
    }

    return (
        <>
            <div
                className={styles.accountWrapper}
                onClick={() => setVisibility(!isVisible)}
            >
                <div className="fas fa-user-circle"></div>
                <span>Hi, {formatName(userData.name)}!</span>
            </div>
        </>
    );
};

export default Account;
