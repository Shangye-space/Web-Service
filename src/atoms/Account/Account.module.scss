@import '../../settings/variables.scss';

.accountWrapper {
    display: block;
    color: $light-grey;

    font-size: 28px;

    & :hover {
        color: $soft-orange;
        cursor: pointer;
    }
}
