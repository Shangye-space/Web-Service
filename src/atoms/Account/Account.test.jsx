import React from 'react';
import ReactDOM from 'react-dom';
import Account from './Account.jsx';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Account userData={{ name: 'Test' }} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(<Account userData={{ name: 'Test' }} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
