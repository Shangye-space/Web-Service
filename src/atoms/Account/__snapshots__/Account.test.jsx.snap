// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<div
  className="accountWrapper"
  onClick={[Function]}
>
  <div
    className="fas fa-user-circle"
  />
  <span>
    Hi, 
    Test
    !
  </span>
</div>
`;
